temporalio/worker/workflow_sandbox/_runner.py

```diff
@@ -112,6 +113,8 @@ def __init__(
         self.runner_class = runner_class
         self.importer = Importer(restrictions, RestrictionContext())
 
+        self._current_thread_id: int | None = None  # Optional[int]
```

Comment:
Why not just use `Optional[int]` here and below to match the original type signature rather than the union?